### ISSUE TITLE
azurerm_recovery_services_vault - soft_delete_enabled not setting to false

### ISSUE THREAD

**Issue by senterfd-jrg** [hashicorp/terraform-provider-azurerm#15675]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_2.98.0`

### Terraform Configuration Files

```hcl
resource "azurerm_recovery_services_vault" "backup" {

  name                = var.recovery_services_vault_name
  resource_group_name = azurerm_resource_group.backup_rg.name
  location            = azurerm_resource_group.backup_rg.location
  sku                 = "Standard"

  soft_delete_enabled = false

  identity {
    type = "SystemAssigned"
  }

  encryption {
    key_id = azurerm_key_vault_key.des.id
    infrastructure_encryption_enabled = false
  }
}
```

### Expected Behaviour

Terraform should properly create the new azurerm_recovery_services_vault resource with `soft_delete_enabled = false`

### Actual Behaviour

Terraform created the RSV but the value of soft_delete_enabled is `true` instead of the intended value `false`. When I try to re-run my TF scripts, I am informed with the information below notifying me the resource needs updating. If I manually change the resource's value using the Azure Portal, this update is not required.
```hcl
  # azurerm_recovery_services_vault.backup will be updated in-place
  ~ resource "azurerm_recovery_services_vault" "backup" ***
        id                  = "<some id>"
        name                = "<some name>"
      ~ soft_delete_enabled = true -> false
        tags                = ***
        # (4 unchanged attributes hidden)
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- Running in Azure Government - Texas region.

### References

* ArmyEITaaS/Issues/issues/534

**Comment by myc2h6o** [comment]
I happened to try on something similar and I'm able to set `soft_delete_enabled` to `false` in public cloud. Maybe there is an issue setting it in the location `usgovtexas`?

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "yicma-rg"
  location = "eastus"
}

resource "azurerm_recovery_services_vault" "test" {
  name                = "yicma-recovery-vault"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku                 = "Standard"
  
  soft_delete_enabled = false
  encryption {
    key_id                            = "https://yicmakv042702.vault.azure.net/keys/k0"
    use_system_assigned_identity      = true
    infrastructure_encryption_enabled = false
  }

   identity {
    type = "SystemAssigned"
  }
}

```

**Comment by sinbai** [comment]
@myc2h6o Yes,exactly, thank you. It could not be reproduced now. However,  I was able to reproduce this issue a few days ago in the location eastus. It is possible that the service backend has changed. Whatever, @senterfd-jrg Could you verify whether you still have this issue?

**Comment by myc2h6o** [comment]
Got another idea that this might be because of the missing Long-Running-Operation earlier in the v2.89.0. Issue has been fixed in v3.0.0 by #15877, maybe @senterfd-jrg you can give it a try

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
A related PR (#15877) has been merged and was released in v3.0.0.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
